### DISCUSSION TITLE
Update ProjectTracker for CSLA 8

### DISCUSSION THREAD

**Discussion by rockfordlhotka** [MarimerLLC/csla#3689] [General]
I have failed to update ProjectTracker for some years now - shame on me!

I started to go down this road again - thinking I'd release CSLA 8 and then do ProjectTracker. Thankfully I brought this idea up on the Discord server, and was called out on my b.s.

So instead, I've decided to update ProjectTracker, but shrink the scope to something more realistic - which is to have only one UI: Blazor.

I'll add other UIs over time - but the big thing is to get the business and DAL layers updated and working with modern CSLA 8, and to have _some_ UI to show how it works, and Blazor is my default UI tech for nearly everything at the moment.

So, here's the PR where I'm doing the work: https://github.com/MarimerLLC/csla/pull/3688

Today I removed all the UI projects, updated the app server to .NET 8, and got the business library to compile. I'm sure it isn't _correct_, but it now compiles. So I have a reasonable start point on which to build.

And yes, @joshhanson314, that ban hammer is still floating out there!! 😀

**Comment by joshhanson314** [comment]
I am dying laughing. I and the world's weirdest celebrity now.

**Comment by Chicagoan2016** [comment]
@rockfordlhotka 
I am looking at one of the properties in ProjectEdit.cs 
```cs
    public static readonly PropertyInfo<string> NameProperty = 
      RegisterProperty<string>(nameof(Name));
    [Display(Name = "Project name")]
    [Required]
    [StringLength(50)]
    public string Name
    {
      get { return GetProperty(NameProperty); }
      set { SetProperty(NameProperty, value); }
    }
```
This property is decorated with some nifty attributes, do you use code generation?

Kind regards

**Reply by rockfordlhotka** [reply]
No, those attributes are from `System.ComponentModel.DataAnnotations` and have been around since around .NET 3.5.

The CSLA rules engine elevates them to "actual rules" so they run within the context of the rules engine - potentially in addition to running in specific UI technologies like ASP.NET MVC.

The attributes are useful in single tenant scenarios. Not so useful in a multi-tenant server scenario because they are hard-coded into the app.

**Reply by Chicagoan2016** [reply]
I saw the attribute  [StringLength(50)] and I thought it was generated by a tool : ). In the past I have worked with a tool that would generate something similar based on sql column, such as varchar(50)
